extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/QueryChangeLogListener.java

```diff
@@ -0,0 +1,16 @@
+package org.apache.rya.streams.api.queries;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Listener to be notified when {@link QueryChange}s occurr on a {@link QueryChangeLog}
+ */
+@DefaultAnnotation(NonNull.class)
+public interface QueryChangeLogListener {
+    /**
+     * Notifies the listener that a query change event has occurred in the change log.
+     * @param queryChangeEvent - The event that occurred. (not null)
+     */
+    public void notify(final ChangeLogEntry<QueryChange> queryChangeEvent);
```

Comment:
Maybe indicate potentially long lasting operations should not be performed within this method. As notification is implemented within this review, everything that tries to access the repository will be blocked while a listener is handling the notification. If that listener is blocked waiting for a timeout or something while submitting a query for execution, it'll lock everybody out of the repository object until that job is finished.